### DISCUSSION TITLE
Issue with people search

### DISCUSSION THREAD

**Discussion by manjunathreddy1** [microsoft-search/pnp-modern-search#3265] [General]
Hi,

We are using the pnp search webparts 4.8 for people search in email value is not coming up from azure ad even though values are present.

Please let us know what we can do to resolve this issue.

Manjunath

**Comment by kasperbolarsen** [comment]
It is very hard to assist you without seeing you current configuration, pics would be nice.

First switch to the debug layout, and check if workemail shows up 

Most likely it will not, which indicates that the workemail property is not in the Selected Properties list, which is required.

**Comment by manjunathreddy1** [comment]
> It is very hard to assist you without seeing you current configuration, pics would be nice.
> 
> First switch to the debug layout, and check if workemail shows up
> 
> Most likely it will not, which indicates that the workemail property is not in the Selected Properties list, which is required.

I added work email field but still the values are not coming. 
![image](https://github.com/microsoft-search/pnp-modern-search/assets/265137/3a2128b0-4491-4481-8d39-552299d7f4d7)
result
![image](https://github.com/microsoft-search/pnp-modern-search/assets/265137/6b902c9c-2064-4af2-ac1f-bac706a83b85)

**Reply by kasperbolarsen** [reply]
Are you using Microsoft Search as the data source?

**Comment by kasperbolarsen** [comment]
At the moment Microsoft people search is in beta and should in my opinion not be used for anything but testing.

However when using SharePoint Search:
Doh, it has been to long since I worked with the People layout.
The property mapped to the email in the layout is UserName
![image](https://github.com/microsoft-search/pnp-modern-search/assets/20593570/033aaa76-6cc0-4805-ac83-45afb2eed5da)

Please verify that the Result Source ID is set to LocalPeopleResults and that Seleceted 
![image](https://github.com/microsoft-search/pnp-modern-search/assets/20593570/2d56ec5c-662f-4e8c-aec6-5e1a72062cdc)


next step should be to verify the raw data, please use one of the tools from here: https://microsoft-search.github.io/pnp-modern-search/using-query-tools-to-verify-issues/ and verify that the WorkEmail property actually contains valid data.

**Reply by manjunathreddy1** [reply]
Thanks for the reply. Not clear how Microsoft people search is in beta! even Microsoft is using graph apis for people search right. Please help to get clarification.

**Reply by kasperbolarsen** [reply]
MS might be cheating, using a private API, perhaps ;-)
You should change to SharePoint as the Datasource for now and perhaps follow the walkthrough here for a quick starter, https://microsoft-search.github.io/pnp-modern-search/scenarios/Create-a-useful-People-Search/

**Reply by kasperbolarsen** [reply]
Try to search the MS People search for *  , it doesn't work, and I reported it to MS about 4-5 months ago

**Reply by wobba** [reply]
As for what Microsoft does internally for people search, that is a tad more complicated than "using Graph" 🙂 For API scenarios I'm with @kasperbolarsen still...use SharePoint APi for people search until Graph have parity. And if you read Graph docs, you see what is returned is a subset of people, not everyone in AAD. See https://learn.microsoft.com/en-us/graph/search-concept-person.
